console/src/telemetry/index.ts

```diff
@@ -0,0 +1,109 @@
+import endpoints from '../Endpoints';
+import globals from '../Globals';
+import { filterEventsBlockList, sanitiseUrl } from './filters';
+import { RUN_TIME_ERROR } from '../components/Main/Actions';
+import { REDUX_LOCATION_CHANGE_ACTION_TYPE } from '../constants';
+
+type ReduxAction = {
+  type: string;
+  payload: {
+    pathname: string;
+  };
+  data: any;
+};
+
+interface TelemetryGlobals {
+  serverVersion: string;
+  consoleMode: string;
+  cliUUID: string;
+  hasuraUUID: string;
+}
+
+const createClient = () => {
+  if (globals.enableTelemetry) {
+    try {
+      const client = new WebSocket(endpoints.telemetryServer);
+      client.onerror = e => {
+        console.error(`WebSocket Error for Events${e}`);
+      };
+      return client;
+    } catch (e) {
+      console.error('Unable to initialise telemetry client', e);
+      return null;
+    }
+  }
+  return null;
+};
+
+let client = createClient();
+if (client) {
+  const onClose = () => {
+    client = createClient();
+    if (client) {
+      client.onclose = onClose;
+    }
+  };
+  client.onclose = onClose;
+}
+
+const isTelemetryConnectionReady = () => {
+  return !!(client && client.readyState === client.OPEN);
+};
+
+const sendEvent = (payload: any) => {
+  if (client && isTelemetryConnectionReady()) {
+    client.send(
+      JSON.stringify({ data: payload, topic: globals.telemetryTopic })
+    );
+  }
+};
+
+export const trackReduxAction = (action: ReduxAction, getState: any) => {
```

Comment:
Maybe we can say at least that this `getState` is a function that returns an object?
`getState: () => Record<string, any>`